express-notes-api/server.ts

```diff
@@ -0,0 +1,131 @@
+import { readFile, writeFile } from 'node:fs/promises';
+import express from 'express';
+
+const app = express();
+
+app.use(express.json());
+
+app.get('/api/notes', async (req, res) => {
+  try {
+    const notesArray = await read();
+    res.status(200).json(notesArray);
+  } catch (error) {
+    res.status(404).json({ error: `${error.message}` });
+  }
+});
+
+app.get('/api/notes/:id', async (req, res) => {
+  try {
+    const { notes } = await readData();
+    const requestId = Number(req.params.id);
+    if (requestId < 0) {
+      res.status(400).json({ error: 'ID needs to be a positive number' });
+      return;
+    }
+    if (!notes[requestId]) {
+      res.status(404).json({ error: `ID ${requestId} not in notes` });
+      return;
+    }
+    res.status(200).json(notes[requestId]);
+  } catch (error) {
+    res.status(404).json({ error: `${error.message}` });
+  }
+});
+
+app.post('/api/notes', async (req, res) => {
+  try {
+    let { nextId, notes } = await readData();
+    const { content } = req.body;
+    if (!content) {
+      res.status(400).send({ error: 'content is a required field' });
```

Comment:
Use `.json` to send a JSON object back. It turns out Express will do that for you since you are sending an object to `.send`, but don't rely on that. Make it explicit what you are doing.

Reply:
:+1: